Marlin/src/HAL/HAL_STM32/pinsDebug_STM32duino.h

```diff
@@ -0,0 +1,276 @@
+ *  Three arrays are used.
+ *
+ *  digitalPin[] is provided by the platform.  It consists of the Port_pin numbers in
+ *  Arduino pin number order.
+ *
+ *  pin_array is a structure generated by the pins/pinsDebug.h header file.  It is generated by
+ *  the preprocessor. Only the signals associated with enabled options are in this table.
+ *  It contains:
+ *    - name of the signal
+ *    - the Ard_num assigned by the pins_YOUR_BOARD.h file using the platform defines.
+ *        EXAMPLE:  "#define KILL_PIN  PB1" results in Ard_num of 57.  57 is then used as an
+ *                  index into digitalPin[] to get the Port_pin number
+ *    - if it is a digital or analog signal.  PWMs are considered digital here.
+ *
+ *  pin_xref is a structure generated by this header file.  It is generated by the
+ *  preprocessor. It is in port/pin order.  It contains just the port/pin numbers defined by the
+ *  platform for this variant.
+ *    - Ard_num
+ *    - printable version of Port_pin
+ *
+ *  Routines with an "x" as a parameter/argument are used to search the pin_array to try to
+ *  find a signal name associated with a port/pin.
+ *
+ *  NOTE -  the Arduino pin number is what is used by the M42 command, NOT the port/pin for that
+ *          signal.  The Arduino pin number is listed by the M43 I command.
+ */
+
+extern const PinName digitalPin[];  // provided by the platform
+
+////////////////////////////////////////////////////////
+//
+// make a list of the Arduino pin numbers in the Port/Pin order
+//
+
+#define _PIN_ADD_2(NAME_ALPHA, ARDUINO_NUM) { {NAME_ALPHA}, ARDUINO_NUM },
+#define _PIN_ADD(NAME_ALPHA, ARDUINO_NUM) { NAME_ALPHA, ARDUINO_NUM },
+#define PIN_ADD(NAME) _PIN_ADD(#NAME, NAME)
+
+typedef struct {
+  char Port_pin_alpha[5];
+  pin_t Ard_num;
+} XrefInfo;
+
+const XrefInfo pin_xref[] PROGMEM = {
+  #include "pins_Xref.h"
+};
+
+////////////////////////////////////////////////////////////
+
+#define MODE_PIN_INPUT  0 // Input mode (reset state)
+#define MODE_PIN_OUTPUT 1 // General purpose output mode
+#define MODE_PIN_ALT    2 // Alternate function mode
+#define MODE_PIN_ANALOG 3 // Analog mode
+
+#define PIN_NUM(P) (P & 0x000F)
+#define PIN_NUM_ALPHA_LEFT(P) (((P & 0x000F) < 10) ? ('0' + (P & 0x000F)) : '1')
+#define PIN_NUM_ALPHA_RIGHT(P) (((P & 0x000F) > 9)  ? ('0' + (P & 0x000F) - 10) : 0 )
+#define PORT_NUM(P) ((P  >> 4) & 0x0007)
+#define PORT_ALPHA(P) ('A' + (P  >> 4))
+
+/**
+ * Translation of routines & variables used by pinsDebug.h
+ */
+#define NUMBER_PINS_TOTAL NUM_DIGITAL_PINS
+#define VALID_PIN(ANUM) ((ANUM) >= 0 && (ANUM) < NUMBER_PINS_TOTAL)
+#define digitalRead_mod(Ard_num) extDigitalRead(Ard_num)  // must use Arduino pin numbers when doing reads
+#define NAME_FORMAT(p) PSTR("%-##p##s")
+#define PRINT_PIN(Q)
+#define PRINT_PORT(ANUM) port_print(ANUM)
+#define DIGITAL_PIN_TO_ANALOG_PIN(ANUM) -1  // will report analog pin number in the print port routine
+#define GET_PIN_MAP_PIN_M43(Index) pin_xref[Index].Ard_num
```

Comment:
I assume we don't need to use the special `PROGMEM` accessors on the STM32 platform…?